pages/katsu/configuration.mdx

```diff
@@ -0,0 +1,313 @@
+
+A path to a file containing the script to run, can be replaced with `inline` (optional)
+
+#### `scripts.*.[].id`
+
+A unique identifier for the script, used for dependency trees, similar to how Ansible works (optional)
+
+#### `scripts.*.[].needs`
+
+A list of script IDs that must be run before this script (optional)
+
+#### `scripts.*.[].name`
+
+A name for the script, used for logging (optional)
+
+---
+
+Files should be placed in the same directory as the manifest file that loads it.
+So if the manifest that loads `hello-world-file` is in `modules/hello.yaml`, the file should be in `modules/hello-world.sh`
+
+---
+
+The best practice is to create a separate folder for each module, and place the required files in that folder.
+
+### `dnf`
+
+This section configures the DNF package manager.
+
+```yaml
+dnf:
+  releasever: 39
+  repodir: /etc/yum.repos.d
+  options:
+    - --setopt=install_weak_deps=False
+    - --nogpgcheck
+  exclude:
+    - foo
+    - bar
+  packages:
+    - baz
+
+  arch_packages:
+    x86_64:
+      - baz-x86_64
+    aarch64:
+      - baz-aarch64
+```
+
+#### `dnf.releasever`
+
+The release version DNF should pull repos from, replaces DNF's `$releasever` variable.
+
+#### `dnf.repodir`
+
+The directory to load repositories from.
+
+#### `dnf.options`
+
+A list of additional arguments to pass to DNF.
+
+#### `dnf.exclude`
+
+A list of packages to exclude from the image.
+
+#### `dnf.packages`
+
+A list of packages to install in the image.
+This can also be any DNF spec string that DNF understands, such as `@core` or `foo.x86_64`
+
+#### `dnf.arch_packages`
+
+A map of packages to install for each architecture. The key is the architecture, and the value is a list of packages to install.
+
+### `disk`
+
+This section configures the disk layout (GPT only)
+
+```yaml
+disk:
+  size: 8GiB
+  partitions:
+    - label: EFI
+      size: 512MiB
+      filesystem: efi
+      mountpoint: /boot/efi
+
+    - label: boot
+      size: 1GiB
+      filesystem: ext4
+      mountpoint: /boot
+
+    - label: root
+      filesystem: ext4
+      mountpoint: /
+```
+
+#### `disk.size`
+
+The total size of the disk image, defined by [bytesize]
+
+#### `disk.partitions`
+
+A list of partitions to create on the disk.
+
+##### `disk.partitions.[].label`
+
+The label of the partition
+
+##### `disk.partitions.[].size`
+
+The size of the partition, defined by [bytesize]
+
+##### `disk.partitions.[].filesystem`
+
+The filesystem to format the partition with `mkfs`
+
+##### `disk.partitions.[].mountpoint`
+
+The mountpoint to mount the partition at relative to the root of the image.
+
+---
+
+The order of the partitions is important, as they are created in the order they are defined, however, mountpoints will be sorted
+by nesting level, so `/` will be mounted first, then `/boot`, then `/boot/efi` This is done using a topological sort algorithm.
+
+If the mountpoint depths are the same, they will be sorted alphabetically.
+
+---
+
+### `import.[]`
```

Comment:
Might want to also state that not all attributes are going to be merged / have the list here

https://github.com/FyraLabs/katsu/blob/main/src/config.rs#L154

Reply:
Look better?